spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/ConverterAwareMappingDatastoreEntityConverter.java

```diff
@@ -71,7 +73,11 @@
 
 	private final GenericConversionService conversionService = new DefaultConversionService();
```

Comment:
I think this looks rather confusing when finally seeing what using 2-layers of converters actually looks like.

There are now 4 things to somebody not deeply-in-the-know that look nearly interchangeable:

```
private final GenericConversionService conversionService
private final GenericConversionService toNativeConversionService
private final DatastoreSimpleTypes toNativeConversions
private final DatastoreCustomConversions customConversions
```

I much prefer reading and understanding the previous version where we don't use the 2-layer shoe-horned Spring Data Core custom conversions. It's also harder to reason about if our test cases are comprehensive enough when dealing with this multi-layer conversion.

At the end of the day if somebody is switching from Mongo to Datastore , a whole different storage tech, is that they need to supply a Double->Long converter rather than the Double->Int converter really a material inconvenience at all?